### PR TITLE
[Performance](Nereids): add withGroupExprLogicalPropChildren to reduce new Plan

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

Issue Number: close #xxx

use `withGroupExprLogicalPropChildren` instead of `withChildren().withGroupExpr().withLogicalPropertiy()`

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.3 seconds
 stream load tsv:          457 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          68.9 seconds inserted 10000000 Rows, about 145K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705054855_clickbench_pr_172497.html

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.83 seconds
 stream load tsv:          462 seconds loaded 74807831229 Bytes, about 154 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          67.7 seconds inserted 10000000 Rows, about 147K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705075043_clickbench_pr_172571.html

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new mechine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.63 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         18 seconds loaded 2358488459 Bytes, about 124 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          88.8 seconds inserted 10000000 Rows, about 112K ops/s
 storage size: 17162378321 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705190246_clickbench_pr_172751.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.41 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          68.2 seconds inserted 10000000 Rows, about 146K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705124554_clickbench_pr_172877.html

**Review by starocean999 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalAggregate.java [hunk 1]
```diff
@@ -256,6 +256,13 @@ public LogicalAggregate<Plan> withLogicalProperties(Optional<LogicalProperties>
                 hasPushed, sourceRepeat, Optional.empty(), logicalProperties, children.get(0));
     }
 
+    @Override
+    public Plan withGroupExprLogicalPropChildren(Optional<GroupExpression> groupExpression,
+            Optional<LogicalProperties> logicalProperties, List<Plan> children) {
+        return new LogicalAggregate<>(groupByExpressions, outputExpressions, normalized, ordinalIsResolved, generated,
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalAggregate.java:262]: u should check children size when replace children

**Review by morrySnow — APPROVED** [review]

**Pull request merged**